kernel/src/igvm_params.rs

```diff
@@ -348,10 +349,28 @@ impl IgvmParams<'_> {
             ));
         }
 
-        regions.push(MemoryRegion::new(
-            PhysAddr::new(self.igvm_param_block.firmware.start as usize),
-            self.igvm_param_block.firmware.size as usize,
-        ));
+        let fw_region =
+            MemoryRegion::new(PhysAddr::new(fw_info.start as usize), fw_info.size as usize);
+
+        // If this firmware expects an IGVM memory map but the IGVM memory
+        // map is not within the firmware GPA range, then add the IGVM memory
+        // map to the set of firmware regions.
+        if fw_info.memory_map_page_count != 0 {
+            let map_region = MemoryRegion::new(
+                PhysAddr::from(fw_info.memory_map_page as u64 * PAGE_SIZE as u64),
+                fw_info.memory_map_page_count as usize * PAGE_SIZE,
+            );
+            if !fw_region.contains_region(&map_region) {
```

Comment:
How about something like the following to make it more generic? Technically speaking the IGVM memory map region shouldn't overlap with any of the fw regions. Of course this would require `fw_region` to be pushed into `regions` prior to the `memory_map_page_count` check.
```suggestion
            if let Some(region) = regions.iter().find(|mr| mr.contains_region(&map_region)) {
```